### PR TITLE
Functional search

### PR DESCRIPTION
**Pull request by arrooney**
## Search bubble appears at the top of each screen:
image
image
image

## Search screen (no query)
image

## Search screen (one tag)
image

## Search screen (multiple tags)
image

closes issue #28 
Closes issue #22

### CODE REVIEW

File: app/src/main/java/com/T05/krowdtrialz/ui/search/SearchActivity.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public boolean onQueryTextChange(String newText) {
+        searchExperiments(newText);
+        return true;
+    }
+
+    public void searchExperiments(String searchString) {
+        experimentAdapter.clear();
+
+        Log.e(TAG, "Query: " + searchString);
+        db.getExperimentsByTags(searchString, new Database.QueryExperimentsCallback() {
```
Comment (Ricky-Au) [inline app/src/main/java/com/T05/krowdtrialz/ui/search/SearchActivity.java:73]: getExperimentsByTags takes in a ArrayList<String> for the first parameter where we split the search string by spaces. I think this isn't the most up to date file.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -9,7 +9,9 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.KrowdTrialz">
-        <activity android:name=".ui.SearchActivity"></activity>
+        <activity android:name=".ui.search.SearchActivity"
+            android:windowSoftInputMode="adjustNothing">
```
Comment (arrooney) [inline app/src/main/AndroidManifest.xml:13]: This was so the on-screen keyboard doesn't squish the UI

**Comment by arrooney** [comment]
@rpshukla the tests are broken. I think we need a way to mock SharedPreferences

File: app/src/main/java/com/T05/krowdtrialz/MainActivity.java [hunk 3]
```diff
@@ -30,6 +38,16 @@ protected void onCreate(Bundle savedInstanceState) {
         NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
         NavigationUI.setupWithNavController(navView, navController);
 
+        final FloatingActionButton openSearchView = (FloatingActionButton) findViewById(R.id.search_action_button);
+        openSearchView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.d("SUBSCRIBED FRAGMENT", "Clicked Search");
```
Comment (vasu-gupta01) [inline app/src/main/java/com/T05/krowdtrialz/MainActivity.java:45]: Not a biggy, but should we have a `private static final String TAG` for this.

**Comment by rpshukla** [comment]
> @rpshukla the tests are broken. I think we need a way to mock SharedPreferences

Hmm maybe we could use something like Mockito to mock it? We could have the mock return a hardcoded id.

**Pull request merged**